internal/policy/operator/restricted_network_aware.go

```diff
@@ -0,0 +1,99 @@
+package operator
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/operator-framework/api/pkg/manifests"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/check"
+	libcsv "github.com/redhat-openshift-ecosystem/openshift-preflight/internal/csv"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/image"
+)
+
+var _ check.Check = &FollowsRestrictedNetworkEnablementGuidelines{}
+
+type FollowsRestrictedNetworkEnablementGuidelines struct{}
+
+func (p FollowsRestrictedNetworkEnablementGuidelines) Validate(ctx context.Context, imgRef image.ImageReference) (bool, error) {
+	return p.validate(ctx, imgRef.ImageFSPath)
+}
+
+//nolint:unparam // ctx is unused. Keep for future use.
+func (p FollowsRestrictedNetworkEnablementGuidelines) getBundleCSV(ctx context.Context, bundlepath string) (*operatorsv1alpha1.ClusterServiceVersion, error) {
+	bundle, err := manifests.GetBundleFromDir(bundlepath)
+	if err != nil {
+		return nil, err
+	}
+	return bundle.CSV, nil
+}
+
+func (p FollowsRestrictedNetworkEnablementGuidelines) validate(ctx context.Context, bundledir string) (bool, error) {
+	logger := logr.FromContextOrDiscard(ctx)
+	csv, err := p.getBundleCSV(ctx, bundledir)
+	if err != nil {
+		return false, err
+	}
+
+	// If the CSV does not claim to support disconnected environments, there's no reason to check that it followed guidelines.
+	if !libcsv.HasInfrastructureFeaturesAnnotation(csv) {
+		logger.Info("warning: this operator does not have the infrastructure-features annotation, unable to determine if it supports disconnected environments.")
+		return false, nil
+	}
+
+	if !libcsv.SupportsDisconnected(csv.Annotations[libcsv.InfrastructureFeaturesAnnotation]) {
+		logger.Info("warning: the infrastructure-features enabled for this operator did not include the \"disconnected\" identifier.")
+		return false, nil
+	}
+
+	// You must have at least one related image (your controller manager) in order to be considered restricted-network ready
+	if !libcsv.HasRelatedImages(csv) {
+		logger.Info("warning: this operator did not have any related images, and at least one is expected")
+		return false, nil
+	}
+
+	// All related images must be pinned. No tag references.
+	if !libcsv.RelatedImagesArePinned(csv.Spec.RelatedImages) {
+		logger.Info("warning: a related image is not pinned to a digest reference of the same image, and this is required.")
+		return false, nil
+	}
+
+	// Some environment variables should be passed into the environment using the RELATED_IMAGE_ prefix.
+	// This isn't the only way to pass this kind of information to the controller, but it is the suggested way in our
+	// documentation.
+	deploymentSpecs := make([]appsv1.DeploymentSpec, len(csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs))
```

Comment:
You need to add a 0 in the middle. This creates a slice with capacity len(DeploymentSpecs) AND length len(DeploymentSpecs). Since you are using append, you want a capacity of len(DeploymentSpecs) and a length of 0. It probably hasn't caused an issue because the zero value for DeploymentSpec is not causing a problem, but you will end up with a slice that is populated with half zero value DeplymentSpecs at the beginning, and the second half being the ones you are appending.

Reply:
Derp, you're right - should be fixed.